File: 0169-majority-element/0169-majority-element.c
//boyer moore algorithm

int majorityElement(int* nums, int numsSize) {
    int candidate = 0;
    int vote = 0;
    for(int i=0; i<numsSize; i++){
        if(vote==0){
            candidate=nums[i];
        }
        if(nums[i]==candidate){
            vote++;
        }
        else{
            vote--;
        }
    }
    return candidate;
}

File: 0169-majority-element/0169-majority-element.py
class Solution:
    def majorityElement(self, nums: list[int]) -> int:
        vote = 0
        candidate = 0

        for i in range(len(nums)):
            if vote == 0:
                candidate = nums[i]
                vote += 1
                
            elif nums[i]==candidate:
                vote += 1
            else:
                vote -= 1
        return candidate